### DISCUSSION TITLE
Lock icon disapears when running sudo/su etc inside tmux

### DISCUSSION THREAD

**Discussion by wadeduvall** [wezterm/wezterm#4637] [Q&A] [answered]
### What Operating System(s) are you running on?

Linux X11

### Which Wayland compositor or X11 Window manager(s) are you using?

i3/picom

### WezTerm version

20230712-072601-f4abf8fd

### Ask your question!

![Screenshot from 2023-12-02 18-48-02](https://github.com/wez/wezterm/assets/11461783/660110e4-dbee-4c4f-a38e-407be015e407)

Basically, if I run su/sudo in wezterm, I get the lock icon (nice!) but if I run tmux, it reverts to the curser bar (ew!). Have spent way too much time trying to figure it out so figured I'd ask here. Other features seem okay (like == and != get replaced with the correct ligatures). Best guess is it has something to do with my tmux colors or fonts, but I wasn't able to figure out what exactly was going on. Nerdfont continues to work elsewhere inside tmux so...

**Comment by wez** [comment, marked as answer]
wezterm uses the state of the pty that it created to spawn your program into to decide when to show the lock icon.

When you are running a multiplexer inside wezterm, the multiplexer makes its own pty to host the programs it runs, and those ptys are not visible to wezterm, so it cannot show the lock icon.

**Reply by wadeduvall** [reply]
Thanks for the info, at least I can stop pounding my head against it. I see that Navigator.nvim supports WezTerm so I might give the built in muxing a shot.
